kafka/consumer.go

```diff
@@ -530,10 +534,14 @@ func (c *pc) consumeRecords(ftp kgo.FetchTopicPartition) {
 		// only the first record is received.
 		last := -1
 		for i, msg := range ftp.Records {
-			meta := make(map[string]string)
+			meta := make(map[string]string, len(msg.Headers))
```

Comment:
We are adding one more entry later for `metadataCtxPartitionKey` key (L543). Would it make sense to specify cap to be `len(msg.Headers) + 1`?

Reply:
Ah thanks  👍🏼 will do

Reply:
I left it as-is in view of Marc's proposal, sowe don't have an additional entry in the map other than the existing headers